src/components/DetailsModal.tsx

```diff
@@ -0,0 +1,167 @@
+import { useEffect, useState } from "react";
+import { useAuth } from "react-oidc-context";
+
+import {
+  Accordion,
+  Alert,
+  Button,
+  Divider,
+  Modal,
+  ModalBody,
+  ModalFooter,
+  Pill,
+  ProgressTracker,
+} from "@us-gov-cdc/cdc-react";
+
+import { Icons } from "@us-gov-cdc/cdc-react-icons";
+import { IFileSubmission } from "@types";
+import getStatusDisplayValuesById, {
+  StatusDisplayValues,
+} from "src/utils/helperFunctions/statusDisplayValues";
+import jsonPrettyPrint from "src/utils/helperFunctions/jsonPrettyPrint";
+import getSubmissionDetails from "src/utils/api/submissionDetails";
+
+interface PropTypes {
+  submission: IFileSubmission;
+  isModalOpen: boolean;
+  handleModalClose: () => void;
+}
+
+function DetailsModal({
+  submission,
+  isModalOpen,
+  handleModalClose,
+}: PropTypes) {
+  const displayValues: StatusDisplayValues = getStatusDisplayValuesById(
+    submission.status
+  );
+  const auth = useAuth();
+  const [details, setDetails] = useState();
+
+  useEffect(() => {
+    const fetchCall = async () => {
+      const res = await getSubmissionDetails(
+        auth.user?.access_token || "",
+        submission.upload_id
+      );
+
+      if (res.status != 200) return;
+
+      try {
+        // TODO: replace any with correct object structure
```

Comment:
This can probably come out since it's mocked.

Reply:
Agreed, removed.